features/ALB002_Alignment-layout.feature

```diff
@@ -0,0 +1,35 @@
+@implementer-agreement
+@ALB
+Feature: ALB002 - Alignment Layout
+
+  Scenario: Agreement on nested attributes of IfcAlignment
+
+      Given A file with Schema Identifier "IFC4X3"
+
+      Then Each IfcAlignment must nest exactly 1 instance(s) of IfcAlignmentHorizontal
+       And Each IfcAlignment must nest at most 1 instance(s) of IfcAlignmentVertical
+       And Each IfcAlignment must nest at most 1 instance(s) of IfcAlignmentCant  
+      # And Every oriented edge shall be referenced exactly 1 times by the loops of the face
+    
+  Scenario: Agreement on attributes being nested within a decomposition relationship
+
+      Given a file with Schema Identifier "IFC4x3"
+
+      Then Each IfcAlignmentHorizontal must be nested only by 1 IfcAlignment
+       And  Each IfcAlignmentVertical must be nested only by 1 IfcAlignment
+       And  Each IfcAlignmentCant must be nested only by 1 IfcAlignment
+    
+  Scenario: Agreement on constraints of allowed attributes nesting nested attributes of IfcAlignment
+    # @note can probably be formulated better
+
+      Given a file with Schema Identifier "IFC4x3"
+
+      Then Each IfcAlignment must nest only the following entities: IfcAlignmentHorizontal, IfcAlignmentVertical, IfcAlignmentCant, IfcReferent, IfcAlignments
+
+  Scenario: Agreement on structure of alignment segments
+
+      Given a file with Schema Identifier "IFC4x3"
+
+       Then Each IfcAlignmentHorizontal nests a list of IfcAlignmentSegment, each of which has DesignParameters typed as IfcAlignmentHorizontalSegment
```

Comment:
I was working on some 'fail files' and also encountered this. For now I just created a file when the nesting list contains a complete different element. 

There's also a possibility (I think) to pass more identifiers, as briefly discussed last Friday. For example: Given a file with (one of the) Schema Identifier (s) "IFC4X3", "IFC4x3_ADD1". Then you can be very precise and add versions later. I'm still trying to work out how this works exactly (An alternative is Jakub's solution, but both solutions give errors at the moment for me; https://github.com/buildingSMART/ifc-gherkin-rules/commit/45bcfa051139339d1d6fb315526d90ee591f57d0). 
A downside I can think of is that you'll have to manually add each new schema to every scenario (and possibly in every feature files). That will create lot of opportunities for typo's and missed scenario/feature files .. Maybe there is a possibility of creating a list with an overview of schema-rule relationships?

Reply:
A last comment (but maybe mostly to myself, hehe): we'll have to be very precise with comma's and whitespaces as well when we give multiple elements in a single statement. I've tried to split 'Then Each IfcAlignment may nest only the following entities: IfcAlignmentHorizontal, IfcAlignmentVertical (etc..)' and kept running into errors. Turned out, after a minor headache, I forgot the whitespace after the comma of each element and was splitting as str.split(',') instead of str.split(', ').  Can you maybe add this to your 'styling document' @Evandro ?